### PR TITLE
Add `0x` prefix to hex values in API calls

### LINKED ISSUE #1185: Update hex number strings in API +0x prefix

### PR DESCRIPTION
**Pull request by grtlr**
# Description of change

In accordance with the following TIP: https://editor.swagger.io/?url=https://raw.githubusercontent.com/iotaledger/tips/stardust-api/tips/TIP-0025/core-rest-api.yaml

## Links to any relevant issues

Closes #1185.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Comment by Thoralf-M** [comment]
Just got `Error("cannot deserialize basic output: invalid type: string \"10000000\", expected u64", line: 1, column: 496)`, the amount field also needs to be changed to a String

**Comment by grtlr** [comment]
~~Ok cool, I can do that in this PR.~~ Will do it in a new PR.

File: bee-api/bee-rest-api/Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ async-trait = { version = "0.1.51", default-features = false, optional = true }
 bech32 = { version = "0.8.1", default-features = false, optional = true }
 digest = { version = "0.9.0", default-features = false, optional = true }
 futures = { version = "0.3.17", default-features = false, optional = true }
-hex = { version = "0.4.3", default-features = false }
+prefix-hex = { version = "0.2.0", default-features = false }
```
Comment (thibault-martinez) [inline bee-api/bee-rest-api/Cargo.toml]: Order!

File: bee-ledger/Cargo.toml [hunk 2]
```diff
@@ -27,7 +27,7 @@ bytes = { version = "1.0.1", default-features = false, optional = true }
 digest = { version = "0.9.0", default-features = false, optional = true }
 futures = { version = "0.3.17", default-features = false, optional = true }
 hashbrown = { version = "0.12.0", default-features = false, optional = true }
-hex = { version = "0.4.3", default-features = false, optional = true }
+prefix-hex = { version = "0.2.0", default-features = false, optional = true }
```
Comment (thibault-martinez) [inline bee-ledger/Cargo.toml]: Order!

File: bee-ledger/Cargo.toml [hunk 3]
```diff
@@ -51,7 +51,7 @@ workers = [
   "digest",
   "futures",
   "hashbrown",
-  "hex",
+  "prefix-hex",
```
Comment (thibault-martinez) [inline bee-ledger/Cargo.toml]: Order!

File: bee-message/Cargo.toml [hunk 4]
```diff
@@ -26,7 +26,7 @@ bytemuck = { version = "1.8.0", default-features = false }
 derive_more = { version = "0.99.16", default-features = false, features = [ "from", "as_ref", "deref" ] }
 digest = { version = "0.9.0", default-features = false }
 hashbrown = { version = "0.12.0", default-features = false, features = [ "ahash", "inline-more" ] }
-hex = { version = "0.4.3", default-features = false }
+prefix-hex = { version = "0.2.0", default-features = false, features = [ "primitive-types1" ] }
```
Comment (thibault-martinez) [inline bee-message/Cargo.toml]: Order!

File: bee-message/src/address/nft.rs [hunk 5]
```diff
@@ -39,13 +39,15 @@ impl FromStr for NftAddress {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        Ok(NftAddress::from(NftId::from(hex_decode(s)?)))
+        Ok(NftAddress::from(NftId::new(
```
Comment (thibault-martinez) [inline bee-message/src/address/nft.rs]: Use NftId::from_str ?

**Review by thibault-martinez — DISMISSED** [review]

**Review by samuel-rufi — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**